src/device.rs

```diff
@@ -11,6 +11,23 @@ use crate::status::{get_device_status, DeviceStatus};
 use crate::{devfs, sysfs, DeviceError, DeviceResult};
 
 #[derive(Debug, Eq, PartialEq)]
+
+/// Abstraction for a single Furiosa NPU device.
+///
+/// # About Furiosa NPU
+///
+/// A Furiosa NPU device contains a number of cores and offers several ways to
+/// utilize these cores. One can utilize them as individual processing
+/// units, fuse the cores for higher performance, or exploit the multicore
+/// mode with customized `send`/`recv` scheduling.
```

Comment:
아래와 같이 작성해봤습니다. 한번 살펴주시고 반영해주시면 감사하겠습니다.

```suggestion
/// A Furiosa NPU device contains a number of cores and offers several ways called
/// [`DeviceMode`][crate::DeviceMode] to combine multiple cores to a single logical device,
/// as following:
/// * [`Single`][crate::DeviceMode::Single]: A logical device is composed of a single core.
/// * [`Fusion`][crate::DeviceMode::Fusion]: Multiple cores work together as if
///     they were one device. This mode is useful when a DNN model requires
///      much computation power and large memory capacity.
/// * [`MultiCore`][crate::DeviceMode::MultiCore]: A logical device uses multiple cores,
///     each of which communicates to one another through interconnect.
///     In this mode, partitions of a model or multiple models can be pipelined.
```
```